### PR TITLE
Fix transmon Hamiltonians

### PR DESCRIPTION
**Pull request by Qottmann**
In order to allow X and Y rotations, we need the drive Hamiltonian to be

$$ H_d(t) = \Omega(t) \sin(\omega_d t + \phi) Y$$

instead of the previously used $H_d(t) = \Omega(t) (\cos(\omega_d t + \phi) X - \sin(\omega_d t + \phi) Y)$. With this change we are following the widespread conventions in [A Quantum Engineer’s Guide to Superconducting Qubits](https://arxiv.org/pdf/1904.06560v5.pdf), in particular eq. (79) - (89). Note that changing the phase dependence is a **necessity** - and using $Y$ (instead of $X$) is a **convention**.

I am not sure whether we should also change the interaction Hamiltonian to just the XX term, again following the convention of [Fully microwave-tunable universal gates in superconducting qubits with linear couplings and
fixed transition frequencies](https://bpb-us-w2.wpmucdn.com/campuspress.yale.edu/dist/2/3627/files/2020/11/Rigetti_Devoret_-_Fully_Microwave-Tunable_Universal_Gates_in_Superconducting_Qubits_with_Linear_Couplings_and_Fixed_Transition_Frequencies.pdf). On IBM devices they describe their interactions via $J_{qp} \sigma^+_q \sigma^-_p + \text{h.c.}$ which leads to `XX + YY` terms. It is _probably_ just a convention. But before changing anything I would leave it as is (with XX + YY).

image
(When you access the ibmq devices you can ask for a hamiltonian description and this is what you find)

### CODE REVIEW

File: pennylane/pulse/transmon.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def transmon_interaction(
         + \sum_{(i, j) \in \mathcal{C}} g_{ij} \left(b^\dagger_i b_j + b_j^\dagger b_i \right)
         + \sum_{q\in \text{wires}} \alpha_q b^\dagger_q b^\dagger_q b_q b_q
 
-    where :math:`[b^\dagger_p, b_q] = i \delta_{pq}` are creation and annihilation operators.
+    where :math:`[b_p, b_q^\dager] = \delta_{pq}` are creation and annihilation operators.
```
Comment (dwierichs) [inline pennylane/pulse/transmon.py:60]: Was this a sign error combined with a single typo (including the `i`)?
```suggestion
    where :math:`[b_p, b_q^\dagger] = \delta_{pq}` are creation and annihilation operators.
```

File: pennylane/pulse/transmon.py [hunk 2]
```diff
@@ -257,23 +257,30 @@ def transmon_drive(amplitude, phase, freq, wires, d=2):
 
     .. math::
 
-        \Omega(t) \left(e^{i (\phi(t) + \nu t)} b_q + e^{-i (\phi(t) + \nu t)} b^\dagger_q\right)
+        \Omega(t) \sin\left(\phi(t) + \nu t\right) \sum_q Y_q
 
-    where :math:`[b^\dagger_p, b_q] = i \delta_{pq}` are creation and annihilation operators
-    and :math:`q` is the qubit label (``wires``).
+    where :math:`Y_q` are Pauli-Y operators on ``wires`` :math:`\{q\}`.
```
Comment (dwierichs) [inline pennylane/pulse/transmon.py:262]:
```suggestion
    where :math:`Y_q` is the Pauli-Y operator on wire :math:`q`.
```

File: pennylane/pulse/transmon.py [hunk 3]
```diff
@@ -257,23 +257,30 @@ def transmon_drive(amplitude, phase, freq, wires, d=2):
...
     The arguments ``amplitude``, ``phase`` and ``freq`` correspond to :math:`\Omega / (2\pi)`, :math:`\phi`
     and :math:`\nu / (2\pi)`, respectively, and can all be either fixed numbers (``float``) or depend on time
     (``callable``). If they are time-dependent, they need to abide by the restrictions imposed
     in :class:`ParametrizedHamiltonian` and have a signature of two parameters, ``(params, t)``.
 
+    Together with the qubit :math:`Z` terms in :func:`transmon_interaction`, driving with this term can generate
+    :math:`X` and :math:`Y` rotations by setting :math:`\phi` accordingly and driving on resonance
+    (see eqs. (79) - (92) in `1904.06560 <https://arxiv.org/abs/1904.06560>`_).
+    Further, it can generate entangling gates by driving at cross-resonance with a coupled qubit
+    (see eqs. (131) - (137) in `1904.06560 <https://arxiv.org/abs/1904.06560>`_). Such a coupling is described in :func:`transmon_interaction`.
```
Comment (dwierichs) [inline pennylane/pulse/transmon.py:272]:
```suggestion
    (see eqs. (131) - (137) in `1904.06560 <https://arxiv.org/abs/1904.06560>`_).
    Such a coupling is described in :func:`transmon_interaction`.
```

File: pennylane/pulse/transmon.py [hunk 4]
```diff
@@ -390,14 +397,12 @@ def qnode(params):
     # Note that exp(-iw)a* + exp(iw)a = cos(w)X - sin(w)Y for a=1/2(X+iY)
     # We compute the `coeffs` and `observables` of the EM field
     coeffs = [
-        AmplitudeAndPhaseAndFreq(qml.math.cos, amplitude, phase, freq),
         AmplitudeAndPhaseAndFreq(qml.math.sin, amplitude, phase, freq),
```
Comment (dwierichs) [inline pennylane/pulse/transmon.py:400]: This should make black put it in one line :)
```suggestion
        AmplitudeAndPhaseAndFreq(qml.math.sin, amplitude, phase, freq)
```

**Review by dwierichs — COMMENTED** [review]
Nice, thanks @Qottmann ! :tada: 
Looking forward to this functioning version!

Just a minor suggestion regarding the changelog, and one question: I was surprised that there were no numerical or analytical results of any QNodes or so in the tests that had to be updated. I guess we would detect them by test failures anyways, but maybe having a test that compares to hardcoded numbers could be a good addition to the tests at some point :)

**Review by mudit2812 — COMMENTED** [review]
Looks good! I'll be happy to approve once @dwierichs 's comments are addressed 😄

File: pennylane/pulse/transmon.py (hunk 1 above)
Comment (Qottmann) [reply pennylane/pulse/transmon.py:60]: I dont know where this came from, probably just confused and no one noticed 😅

File: pennylane/pulse/transmon.py [hunk 5]
```diff
@@ -257,23 +257,30 @@ def transmon_drive(amplitude, phase, freq, wires, d=2):
 
     .. math::
 
-        \Omega(t) \left(e^{i (\phi(t) + \nu t)} b_q + e^{-i (\phi(t) + \nu t)} b^\dagger_q\right)
+        \Omega(t) \sin\left(\phi(t) + \nu t\right) \sum_q Y_q
```
[the remark this answers is not in the document]
Comment (Qottmann) [reply pennylane/pulse/transmon.py:260]: yep

[the remark this answers is not in the document]
Comment (Qottmann) [reply pennylane/pulse/transmon.py:260]: I think there is hardly any use for actually summing something and this is more a relic from wanting to do drives for rydberg and transmons from the same function. But it doesnt hurt to provide the the possibility to have a drive on more than one qubit

File: pennylane/pulse/transmon.py (hunk 3 above)
Comment (Qottmann) [reply pennylane/pulse/transmon.py:272]: Do you want an explicit line break i.e. also in the rendered docs or just for better readability in the code?

File: pennylane/pulse/transmon.py (hunk 2 above)
Comment (dwierichs) [reply pennylane/pulse/transmon.py:262]: Ah, right, but $Y_q$ is a single operator, no?
Maybe this?
```suggestion
    where :math:`\{Y_q\}` are the Pauli-Y operators on ``wires`` :math:`\{q\}`.
```

File: pennylane/pulse/transmon.py (hunk 3 above)
Comment (dwierichs) [reply pennylane/pulse/transmon.py]: Just for code :)

File: tests/pulse/test_transmon.py [hunk 6]
```diff
@@ -438,6 +424,14 @@ def test_wires_and_connections_and_not_containing_each_other_raise_warning(
             H = qml.pulse.transmon_interaction(qubit_freq, connections, coupling, wires)
             assert H.wires == Wires(["a", "b", "c", "d", "e", "f"])
 
+    def test_functional_form(
+        self,
+    ):
```
Comment (dwierichs) [inline tests/pulse/test_transmon.py:429]:
```suggestion
    def test_functional_form(self):
        """Test that the matrices of the transmon drive and a manually constructed
        operator with the same functional form are equal."""
```

**Review by dwierichs — APPROVED** [review]
Looks good to me, thanks @Qottmann :100: :tada:

**Comment by Qottmann** [comment]
[sc-42727]

**Pull request merged**